container-host-files/etc/scf/config/scripts/forward_logfiles.sh

```diff
@@ -1,175 +1,171 @@
                 exit 0
         fi
 }
 
 # check if more logs to be monitored by rsyslog have come into existence since the last run
 function searchTargetDir {
-        filesAdded=1
-        for file in $1/*
-        do
-                if [ -d $file ]; then
-                        ignored=false
-                        for ignore in $IGNORE_DIR
-                        do
-                                if [[ $file == $RSYSLOG_FORWARDER_WATCH_DIR/$ignore ]]; then
-                                        echo "Ignoring $file directory"
-                                        ignored=true
-                                fi
-                        done;
-                        if [ $ignored == false ]; then
-                                if searchTargetDir $file; then
-                                        filesAdded=0
-                                fi
-                        fi
+        local dir file filesAdded=1
+        local args=( "$1" )
+        for dir in ${IGNORE_DIR} ; do
+                args=( "${args[@]}" '(' -name "${dir}" -a -prune ')' -o )
+        done
+        args=( "${args[@]}" '(' -name '*.log' -a -type f -a -print0 ')' )
+        while IFS= read -r -d '' file ; do
+                TARGET_BASENAME="$(basename "$1" .log)"
+                TARGET_NAME="${RSYSLOG_CONF_DIR}/${RSYSLOG_CONF_PREFIX}-${TARGET_BASENAME}.conf"
+                if checkConfigExists ; then
+                        echo "${TARGET_NAME} exists"
                 else
-                        if [ "${file: -4}" == ".log" ]; then
-                                targetName $file
-                                if checkConfigExists $file; then
-                                        echo $TARGET_NAME exists
-                                else
-                                        echo "Creating $TARGET_NAME"
-                                        createTargetConf $file
-                                        filesAdded=0
-                                fi
-                        fi
+                        echo "Creating ${TARGET_NAME}"
+                        createTargetConf "${file}"
+                        filesAdded=0
                 fi
-        done
-        return $filesAdded
+        done < <(find "${args[@]}")
+        return ${filesAdded}
 }
 
 #Create the rsyslog configuration file inside rsysconf.d
 function createTargetConf {
         # We need to strip leading whitespace introduced by the heredoc (because
         # it doesn't strip leading spaces, just tabs)
-        cat <<-EOF | sed 's@^\s*@@' >${TARGET_NAME}
-            \$InputFileName ${1}
-            \$InputFileTag vcap-${TARGET_BASENAME}
-            \$InputFileStateFile ${TARGET_BASENAME}_state
-            \$InputFileFacility local7
-            \$InputRunFileMonitor
+        sed 's@^\s*@@' >"${TARGET_NAME}" <<-EOF
+                input(type="imfile"
```

Comment:
Is `input(type` proper syntax ?

Reply:
It does seem to work, yes. (The closing parenthesis is a few lines down)

Reply:
Ah 💡. Completely missed that. Ok then, good enough.
(Although I wonder if indenting the remainder would help to make the association).